samcli/lib/build/app_builder.py

```diff
@@ -147,29 +147,37 @@ def update_template(self, template_dict, original_template_path, built_artifacts
             Updated template
         """
 
-        original_dir = os.path.dirname(original_template_path)
+        original_dir = pathlib.Path(os.path.dirname(original_template_path))
 
         for logical_id, resource in template_dict.get("Resources", {}).items():
 
             if logical_id not in built_artifacts:
                 # this resource was not built. So skip it
                 continue
 
+            artifact_dir = pathlib.Path(built_artifacts[logical_id])
+
+            # Default path to absolute path of the artifact
+            store_path = str(artifact_dir)
```

Comment:
If we are going to use Pathlib, lets use pathlib. Changing in the middle to strings is a little confusing while reading.

Reply:
This was done so that output from os.path.relpath does not need to be turned into pathlib.Path again and store_path can be directly stored into properties[].

pathlib.Path.relative_to does not work as it requires one to be a subpath of the other unlike os.path.relpath.